test_app/.gitattributes

```diff
@@ -0,0 +1,10 @@
+# See https://git-scm.com/docs/gitattributes for more about git attribute files.
```

Comment:
What is this file about?

Reply:
Since developers use different operating systems, the default file ending lines will be different. The default on Windows is Carriage Return Line Feed (CRLF), however, on Linux/MacOS it is Line Feed (LF). Here is where and when the gitattributes file comes into rescue.

A gitattributes file is a simple text file that gives attributes to pathnames.
Each line in gitattributes file is of form: `pattern attr1 attr2 ...`

Belows are some examples from gitattributes official documentation for eol and text attributes:
```
*               text=auto #
```
When text attribute is enabled set to "auto", the path is marked for automatic end-of-line conversion. If Git decides that the content is text, then its line endings are automatically converted to LF when pushing to code. When the file has been committed with CRLF, no conversion is done.

```
*.txt		text
*.jpg		-text
```
For files end with .txt, setting the text attribute on a path enables end-of-line normalization and marks the path as a text file. 

For files end with .jpg, unsetting the text attribute on a path tells Git not to attempt any end-of-line conversion upon pushing the changesor pulling the changes.

```
*.vcproj	text eol=crlf
*.sh		text eol=lf
```
Setting the text attribute on a path enables end-of-line normalization and marks the path as a text file. End-of-line conversion takes place without guessing the content type. 

For files end with .vcproj, the eol=crlf setting forces Git to normalize line endings when pushing the code and convert them to CRLF when the file is checked out. For files end with .sh, the eol setting forces Git to normalize line endings to LF when pushing the files the ends with .sh extension and prevents conversion to CRLF when you pull the changes.

Reply:
What does line endings have to do with this file?

Reply:
For example, if you enable the text attributes, like this
`* text=auto` 
the .gitattributes config normalizes line endings to LF for all text files when checked into Git's index. However, the local line endings on your local machine is remaining untouched.